compiler/hash-typecheck/src/options/normalisation.rs

```diff
@@ -8,13 +10,36 @@ use crate::errors::TcError;
 /// A signal which can be emitted during normalisation.
 #[derive(Debug, Clone, From)]
 pub enum NormaliseSignal {
+    /// A `LoopControlTerm::Break` was encountered in a loop.
     Break,
+    /// A `LoopControlTerm::Continue` was encountered in a loop.
     Continue,
+    /// A `ReturnTerm` was encountered in a function, with the given return
+    /// value.
     Return(TermId),
+    /// An typechecking error occurred during normalisation.
     Err(TcError),
 }
 
 /// The result of a normalisation operation.
```

Comment:
It would be nice to perhaps just make a type to represent all these outcomes rather than a newtype wrapping it. But I guess its not really feasible since you can't use `?` on self defined types right?

Reply:
This is why i wrote `@@Improvement: maybe we can make this into a custom enum and implement `std::ops::Try` for it?`

Reply:
Ah right, I didn't quite see it